### DISCUSSION TITLE
Какое максимальное кол-во диодов?

### DISCUSSION THREAD

**Discussion by TheFire580** [vvip-68/LedPanelWiFi#92] [Общие вопросы]
Собрал матрицу 3x2 сегмент 16x16 не работают 2 последние матрицы не работают, все режимы перепробовал

**Comment by vvip-68** [comment]
По отдельности матрицы работают?
В настройках общее количество светодиодов выставлено верно?

Максимально поддерживаемые матрицы - 128x128 - 16384 диодов

**Reply by TheFire580** [reply]
По отдельности работают, каждую проверил, работают, также работают 3x1, 6x1. В плане общее кол-во, в настройках я такого не нашёл, и в коде прошивки тоже, можете подсказать?

**Reply by vvip-68** [reply]
Это все из web-интерфейса настраивается. Примерно здесь:
![image](https://github.com/user-attachments/assets/d3c1fc87-d8d6-481f-b95f-92347a54b992)

**Reply by TheFire580** [reply]
Так же пробовал менять платы, 3 перепробовал одно и тоже, питания от хорошего лабораторника брал.

**Reply by TheFire580** [reply]
Я слепой видимо, ща попробую

**Reply by TheFire580** [reply]
![image](https://github.com/user-attachments/assets/582fdbd8-4612-45b0-b9a0-7b59fd1809f0)
Как понимаю если у меня 1536 диодов всего, то мне надо сплитить на 2 пина?

**Reply by vvip-68** [reply]
А что - на один не влазит?

**Reply by TheFire580** [reply]
ну у меня пишет макс 1024 на один D2

**Reply by vvip-68** [reply]
Покажите предыдущую страницу - Матрица.
У вас там по видимому сконфигурировано 4 матрицы только

**Reply by TheFire580** [reply]
![image](https://github.com/user-attachments/assets/e1d17b8b-e816-4234-bdf4-0051c25e0f5b)

**Reply by vvip-68** [reply]
Это странно. Проверяю.

**Reply by vvip-68** [reply]
Какой-то глюк. Перезагрузите контроллер и обновите страничку в браузере

**Reply by TheFire580** [reply]
Спасибо за помощь, просто заново с гита скачал проект загрузил заработало

**Reply by vvip-68** [reply]
Заново - это правильно. Я сегодня там некоторые изменения добавил. Для надежности типа...

**Reply by vvip-68** [reply]
И на вашем примере с удивлением обнаружил, что Wemos d1 mini тянет 1536 диодов (медленно, правда, но работает). Я думал 600-700 это предел для него... Только на эффекте Мерцание сваливается. Видимо он много памяти требует и ему не хватает...

**Reply by whitedron** [reply]
> И на вашем примере с удивлением обнаружил, что Wemos d1 mini тянет 1536 диодов (медленно, правда, но работает). Я думал 600-700 это предел для него... 

Проверьте пожалуйста в utility2 только 3 пина для подключения ленты инициализируются, хотя назначить можно 4

**Reply by vvip-68** [reply]
Да, спасибо. Действительно, похоже, косяк...

**Comment by 33tooth** [comment]
Здравствуйте, подскажите/уточните, хотел сделать матрицу 50х29 светиков, но Wemos d1 mini потянет ли?

**Reply by vvip-68** [reply]
Пробуйте. Я не знаю. Скомпилируйте, запустите.
Сейчас ESP32 по цене, наверное, чуть ли не дешевле вемосов. Что заставляет вас использовать столь устаревший микроконтроллер? На озоне ESP32-C3 от 270 рублей стоят с доставкой через 5 дней (хотя, они, конечно чаще всего врут со сроками). На алиэкспресс вообще наверное копейки, но там и доставка больше месяца может быть.

**Reply by 33tooth** [reply]
спасибо, да я не уперся в Wemos, как то не думал о других платах. ESP32-C3 интересная плата посмотрел, при прошивке имеет какие то нюансы в скетче? и так же управляется по вайфай из приложения на смарте? сколько светодиодов на этой плате можно забубенить?

**Reply by vvip-68** [reply]
Не помню, чтобы какие-то проблемы были с прошивкой ESP32-C3
В коде ничего менять не нужно - там все учитывается. Просто при прошивке выбрать правильную плату, которая соответствует реальной. Поведение и управление - такое же. Я тестировал на прогоне до 16 тыс светодиодов.

**Reply by 33tooth** [reply]
спасибо, интересная идея, а при прошивке платы кнопки на ней надо нажимать?

**Reply by vvip-68** [reply]
Не помню. Давно это было. А в чем проблема?

**Reply by 33tooth** [reply]
проблем пока нет как и платы, просто на ней две кнопки, вот и интересуюсь как ее прошивают

**Reply by vvip-68** [reply]
Как и все прочие платы с двумя кнопками. Если процесс не стартует автоматически - зажать и держать кнопку BOOT, потом коротко нажать Reset и отпустить BOOT.
